src/main/java/org/basseur/adventofcode/advent2018/days/day08/Node.java

```diff
@@ -0,0 +1,97 @@
+package org.basseur.adventofcode.advent2018.days.day08;
+
+import java.util.Arrays;
+
+public class Node {
+    /** The length of the header the containing amount of child nodes and the length of metadata */
+    private static final int headerLength = 2;
+    /** The metadata of this node */
+    private int[] metadata;
+    /** The child nodes of this node */
+    private Node[] childNodes;
+
+    /**
+     * The constructor parses the nodes array and instantiates its own child nodes, while also storing the meta data
+     *
+     * @param nodes the array of nodes to be parsed
+     */
+    Node(Integer[] nodes) {
```

Comment:
Why Integer and in almost all other cases int? Is there a specific reason here?

Reply:
`fileReaders.readFileIntoArrayOfIntegers()` returns an `Integer[]` which is then passed as an argument to the constructor of `Node`. If I change it, I need to copy the `Integer[]` to `int[]`. I thought that is just one additional and unnecessary step.

Reply:
Sure, alright, just wanted to understand :)